src/middlewared/middlewared/api/base/handler/version.py

```diff
@@ -122,25 +116,28 @@ def adapt_model(self, value: dict, model_name: str, version1: str, version2: str
             value_factory = functools.partial(
                 self._adapt_model, value_factory, model_name, current_version, new_version, direction,
             )
-            model = new_version.models.get(model_name)
+            try:
+                model = await new_version.get_model(model_name)
+            except KeyError:
+                model = None
 
             current_version = new_version
 
-        return model, value_factory()
+        return model, await value_factory()
 
-    def _adapt_model(self, value_factory: Callable[[], dict], model_name: str, current_version: APIVersion,
-                     new_version: APIVersion, direction: Direction):
+    async def _adapt_model(self, value_factory: Callable[[], Awaitable[dict]], model_name: str,
```

Comment:
Please wrap these arguments are single lines like you did for the `def adapt_model` method. (i.e.
```
async def _adap_model(
    self,
    value_factory: Callable...